src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -473,8 +473,30 @@ static bool CallRegisterMethodByIndex (JniNativeMethodRegistrationArguments argu
 		public override void RegisterNativeMembers (JniType nativeClass, Type type, string? methods) =>
 			RegisterNativeMembers (nativeClass, type, methods.AsSpan ());
 
+#if ENABLE_MARSHAL_METHODS
+		// Temporary hack, see comments in RegisterNativeMembers below
+		static readonly Dictionary<string, List<string>> dynamicRegistrationMethods = new Dictionary<string, List<string>> (StringComparer.Ordinal) {
```

Comment:
This would be more efficient if it were a `Dictionary<string, string[]>` instead of a `Dictionary<string, List<string>>`, as the `List<string>` contains a `string[]` anyway…